src/Shell/Task/ModelTask.php

```diff
@@ -253,33 +247,46 @@ public function applyAssociations($model, $associations)
     public function findBelongsTo($model, array $associations)
     {
         $schema = $model->schema();
+        $tables = $this->listAll();
+
         foreach ($schema->columns() as $fieldName) {
             if (!preg_match('/^.*_id$/', $fieldName)) {
                 continue;
             }
 
+            $tmpModelName = $this->_modelNameFromKey($fieldName);
```

Comment:
I might be wrong, but I thought [this if check](https://github.com/ionas/bake/blob/table-belongs-to-composite-primary-keys-new/src/Shell/Task/ModelTask.php#L264-L266) would block `parent_id`.

And reading @thinkingmedia's comment, do you want this feature only in the HABTM associations? Then I would suggest to change only `findBelongsToMany()`. That would make it easy to test, review and merge.